### PR TITLE
Add min/max limits to compensation sensors

### PR DESCRIPTION
**Pull request by Petro31**
## Proposed change

update for https://github.com/home-assistant/core/pull/85886

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/compensation.markdown [hunk 1]
```diff
@@ -63,4 +77,14 @@ unit_of_measurement:
   description: Defines the units of measurement of the sensor, if any.
   required: false
   type: string
+apply_lower_limit:
+  description: "Enables a lower limit for the sensor.  The lower limit is defined by the data collections (`data_points`) lowest `uncompensated_value`.  For example, if the lowest `uncompensated_value` value is `1.0` and the paired `compensated_value` is `0.0`, the any `source` state less than `1.0` will produce a compensated state of `0.0`.""
```
Comment (frenck) [inline source/_integrations/compensation.markdown]:
```suggestion
  description: "Enables a lower limit for the sensor.  The lower limit is defined by the data collections (`data_points`) lowest `uncompensated_value`.  For example, if the lowest `uncompensated_value` value is `1.0` and the paired `compensated_value` is `0.0`, the any `source` state less than `1.0` will produce a compensated state of `0.0`."
```

File: source/_integrations/compensation.markdown [hunk 2]
```diff
@@ -63,4 +77,14 @@ unit_of_measurement:
   description: Defines the units of measurement of the sensor, if any.
   required: false
   type: string
+apply_lower_limit:
+  description: "Enables a lower limit for the sensor.  The lower limit is defined by the data collections (`data_points`) lowest `uncompensated_value`.  For example, if the lowest `uncompensated_value` value is `1.0` and the paired `compensated_value` is `0.0`, the any `source` state less than `1.0` will produce a compensated state of `0.0`."
```
Comment (frenck) [inline source/_integrations/compensation.markdown]:
```suggestion
  description: "Enables a lower limit for the sensor. The lower limit is defined by the data collections (`data_points`) lowest `uncompensated_value`. For example, if the lowest `uncompensated_value` value is `1.0` and the paired `compensated_value` is `0.0`, any `source` state less than `1.0` will produce a compensated state of `0.0`."
```

File: source/_integrations/compensation.markdown [hunk 3]
```diff
@@ -63,4 +77,14 @@ unit_of_measurement:
...
+  required: false
+  type: boolean
+  default: false
+apply_upper_limit:
+  description: "Enables an upper limit for the sensor.  The upper limit is defined by the data collections (`data_points`) greatest `uncompensated_value`.  For example, if the greatest `uncompensated_value` value is `5.0` and the paired `compensated_value` is `10.0`, the any `source` state greater than `5.0` will produce a compensated state of `10.0`."
```
Comment (frenck) [inline source/_integrations/compensation.markdown]:
```suggestion
  description: "Enables an upper limit for the sensor. The upper limit is defined by the data collections (`data_points`) greatest `uncompensated_value`. For example, if the greatest `uncompensated_value` value is `5.0` and the paired `compensated_value` is `10.0`, any `source` state greater than `5.0` will produce a compensated state of `10.0`."
```

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/_integrations/compensation.markdown [hunk 4]
```diff
@@ -30,6 +30,20 @@ compensation:
...
       - [0.2, -80.0]
       - [1.0, 0.0]
+
+  media_player_db_volume:
+    source: media_player.yamaha_receiver_zone_2
+    attribute: volume_level
+    unit_of_measurement: dB
+    # Ensure that the sensor's value will not have a state lower than -80.0
+    # when the source sensors value is less than 0.2
+    apply_lower_limit: True
```
Comment (bdraco) [inline source/_integrations/compensation.markdown]:
```suggestion
    clamp_lower_limit: True
```

File: source/_integrations/compensation.markdown [hunk 5]
```diff
@@ -30,6 +30,20 @@ compensation:
...
+    # Ensure that the sensor's value will not have a state greater than 0.0
+    # when the source sensors value is greater than 1.0
+    apply_upper_limit: True
```
Comment (bdraco) [inline source/_integrations/compensation.markdown]:
```suggestion
    clamp_upper_limit: True
```

File: source/_integrations/compensation.markdown [hunk 6]
```diff
@@ -63,4 +77,14 @@ unit_of_measurement:
   description: Defines the units of measurement of the sensor, if any.
   required: false
   type: string
+apply_lower_limit:
```
Comment (bdraco) [inline source/_integrations/compensation.markdown]:
```suggestion
clamp_lower_limit:
```

File: source/_integrations/compensation.markdown [hunk 7]
```diff
@@ -63,4 +77,14 @@ unit_of_measurement:
...
+  description: "Enables a lower limit for the sensor. The lower limit is defined by the data collections (`data_points`) lowest `uncompensated_value`. For example, if the lowest `uncompensated_value` value is `1.0` and the paired `compensated_value` is `0.0`, any `source` state less than `1.0` will produce a compensated state of `0.0`."
+  required: false
+  type: boolean
+  default: false
+apply_upper_limit:
```
Comment (bdraco) [inline source/_integrations/compensation.markdown]:
```suggestion
clamp_upper_limit:
```

File: source/_integrations/compensation.markdown (hunk 4 above)
Comment (bdraco) [reply source/_integrations/compensation.markdown]: I think there is a enough context to change everything to `lower_limit: True` and `upper_limit: True` which would make the config a bit simplier

File: source/_integrations/compensation.markdown (hunk 7 above)
Comment (Petro31) [reply source/_integrations/compensation.markdown]: Updated to `upper_limit`

File: source/_integrations/compensation.markdown (hunk 6 above)
Comment (Petro31) [reply source/_integrations/compensation.markdown]: updated to `lower_limit`

File: source/_integrations/compensation.markdown (hunk 5 above)
Comment (Petro31) [reply source/_integrations/compensation.markdown]: updated to `upper_limit`

File: source/_integrations/compensation.markdown (hunk 4 above)
Comment (Petro31) [reply source/_integrations/compensation.markdown]: updated to `lower_limit`

**Review by bdraco — DISMISSED** [review]

File: source/_integrations/compensation.markdown [hunk 8]
```diff
@@ -30,6 +30,20 @@ compensation:
...
       - [0.2, -80.0]
       - [1.0, 0.0]
+
+  media_player_db_volume:
+    source: media_player.yamaha_receiver_zone_2
+    attribute: volume_level
+    unit_of_measurement: dB
+    # Ensure that the sensor's value will not have a state lower than -80.0
+    # when the source sensors value is less than 0.2
+    lower_limit: True
```
Comment (frenck) [inline source/_integrations/compensation.markdown]:
```suggestion
    lower_limit: true
```

File: source/_integrations/compensation.markdown [hunk 9]
```diff
@@ -30,6 +30,20 @@ compensation:
...
+    # Ensure that the sensor's value will not have a state greater than 0.0
+    # when the source sensors value is greater than 1.0
+    upper_limit: True
```
Comment (frenck) [inline source/_integrations/compensation.markdown]:
```suggestion
    upper_limit: true
```

**Review by frenck — APPROVED** [review]
Thanks, @Petro31 👍 

../Frenck

**Pull request merged**